src/preprocess/vhi.py

```diff
@@ -0,0 +1,173 @@
+"""
+
+- Add lat lon coordinates
+- add time coordinates
+- subset Kenya
+- merge into one time (~500MB)
+"""
+from pathlib import Path
+import pathlib
+import xarray as xr
+import multiprocessing
+from typing import List, Optional, Tuple, Union
+import pickle
+from functools import partial
+
+from xarray import Dataset
+
+from .base import (BasePreProcessor, get_kenya)
+from .preprocess_vhi import (
+    extract_timestamp,
+    create_lat_lon_vectors,
+    create_new_dataset,
+    create_filename,
+)
+from .preprocess_utils import select_bounding_box_xarray
+
+
+class VHIPreprocessor(BasePreProcessor):
+    """ Preprocesses the VHI data """
+
+    def __init__(self, data_folder: Path = Path('data')) -> None:
+        super().__init__(data_folder)
+
+        self.out_dir = self.interim_folder / "vhi_preprocessed"
+        if not self.out_dir.exists():
+            self.out_dir.mkdir()
+
+        self.vhi_interim = self.interim_folder / "vhi"
+        if not self.vhi_interim.exists():
+            self.vhi_interim.mkdir()
+
+    def get_vhi_filepaths(self) -> List[Path]:
+        return [f for f in (self.raw_folder / "vhi").glob('*/*.nc')]
+
+    @staticmethod
+    def preprocess_VHI_data(netcdf_filepath: str,
+                            output_dir: str,
+                            subset: str = 'kenya',) -> Path:
+        """Run the Preprocessing steps for the NOAA VHI data
+
+        Process:
+        -------
+        * assign time stamp
+        * assign lat lon
+        * create new dataset with these dimensions
+        * Save the output file to new folder
+        """
+        print(f"** Starting work on {netcdf_filepath.split('/')[-1]} **")
+        # 1. read in the dataset
+        ds = xr.open_dataset(netcdf_filepath)
+
+        # 2. extract the timestamp for that file (from the filepath)
+        timestamp = extract_timestamp(ds, netcdf_filepath, use_filepath=True)
+
+        # 3. extract the lat/lon vectors
+        longitudes, latitudes = create_lat_lon_vectors(ds)
+
+        # 4. create new dataset with these dimensions
+        new_ds = create_new_dataset(ds, longitudes, latitudes, timestamp)
+
+        # 5. chop out EastAfrica - TODO: have a dictionary of legitimate args
+        if subset == 'kenya':
+            kenya_region = get_kenya()
+            kenya_ds = select_bounding_box_xarray(new_ds, kenya_region)
+
+        # 6. create the filepath and save to that location
+        filename = create_filename(
+            timestamp,
+            netcdf_filepath,
+            subset=True,
+            subset_name=subset
+        )
+        print(f"Saving to {output_dir}/{filename}")
+        # TODO: change to pathlib.Path objects
+        kenya_ds.to_netcdf(f"{output_dir}/{filename}")
```

Comment:
Since `kenya_ds` is only defined if a subset is defined, this function will fail if a subset isn't defined